internal/cloud/realm/secrets_test.go

```diff
@@ -46,5 +47,28 @@ func TestSecrets(t *testing.T) {
 				assert.Equal(t, []realm.Secret{secret}, secrets)
 			})
 		})
+
+		t.Run("should create secrets for deletion", func(t *testing.T) {
+			testLen := 3
+			testSecrets := make([]realm.Secret, testLen)
+			for i := 0; i < testLen; i++ {
```

Comment:
we probably don't even need all of this here (though it's great testing!).  for the most part, leave the file as is, and just add a few extra tests... it should read something like:

```go
t.Run("should fail without an auth client", func(t *testing.T) {
  
  t.Run("with an active session ", func(t *testing.T) {

    // FIXME: actually we can just delete this test case now, its effectively duplicated with the new delete stuff you'll be adding
    t.Run("should have no secrets upon app initialization", func(t *testing.T) {

    t.Run("should create a secret", func(t *testing.T) {
      t.Run("and list all app secrets", func(t *testing.T) {})
    })

    t.Run("should delete a secret", func(t *testing.T) {

      // TODO: delete the created test secret here

      t.Run("and list all app secrets", func(t *testing.T) {

        // TODO: find all test secrets again and assert it's empty

      })
    })
    })
  })
})
```